### ISSUE TITLE
topk returns different results with the same input in cuda and cpu

### ISSUE THREAD

**Issue by Tengxu-Sun** [pytorch/pytorch#85227]
### 🐛 Describe the bug

torch.topk() has different results for input located at cuda and cpu. To reproduce my error, i'm putting my python script and data at https://github.com/Tengxu-Sun/Pytorch-Topk-Bug.

### Versions

pytorch: 1.10.2
cudakit: 11.3
GPU: 2080 Ti
CPU: Intel(R) Xeon(R) Gold 5120 CPU @ 2.20GHz

According to my test, pytorch 1.9, 1.8, 1.7 and 1.6 also have this error.

**Comment by ezyang** [comment]
1.10 is still pretty old, can you try on 1.12

**Comment by ngimel** [comment]
The topk values are identical, indices are different but that's expected, in case there are several equal values in the topk, the returned index is not guaranteed to match between cpu and cuda.

**Comment by Tengxu-Sun** [comment]
> 1.10 is still pretty old, can you try on 1.12

torch1.12 also have this problem.

**Comment by Tengxu-Sun** [comment]
> The topk values are identical, indices are different but that's expected, in case there are several equal values in the topk, the returned index is not guaranteed to match between cpu and cuda.

Yeah, i do a test. It indeed has lots of equal values. But i wonder why torch1.4 don't have this problem. Torch1.4 produces the same result for this input.
